### PR TITLE
Parsers cleanup

### PR DESCRIPTION
**Pull request by gwiedeman**
## Type of Contribution

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New component
- [X] Refactoring (no functional changes)
- [ ] Documentation-only

## What does this implement/fix? Explain your changes.

Addresses a number of format parser issues. `mailbagit` now handles both directory and individual file input for PSTs and MBOXs. Parsers now return both binary and string bodies to the model (except MSG only provides string) for improved fidelity for derivatives. Model fields now more closely follow the specification (Email_Folder is now Message_Path). Also provides more structured error messages back to the model using a list. Derivatives also minimally normalize file paths from PSTs with special characters, though this needs further improvement.

PDF derivatives still only use string bodies for now, which needs to be addressed.

## Link to issue?

#107 #108 #94 #93 #90 #86

- [X] Issue closed
- [ ] Remain open

## Pull Request Checklist

Please check if your PR fulfills the following requirements:
- [X] Make sure you are requesting to the develop branch. Don't PR to main!
- [ ] This contribution has sufficient documentation
- [ ] Tests for the changes have been added
- [X] All tests pass

#### How has this been tested?
**Operating System:** Win10
**Python Version:** 3.8 and 3.9

## Licensing
- [x] I agree that the Mailbag Project and the University at Albany, SUNY can release this code under the [MIT license](https://github.com/UAlbanyArchives/mailbag/blob/main/LICENSE).

### CODE REVIEW

File: mailbag/formats/mbox.py [hunk 1]
```diff
@@ -30,65 +30,95 @@ def account_data(self):
...
-            subFolder = helper.emailFolder(self.file,filePath)
+        if os.path.isfile(self.file):
+            files = self.file
+            parent_dir = os.path.dirname(self.file)
+        else:
+            files = os.path.join(self.file, "**", "*.mbox")
+            parent_dir = self.file
+        file_list = glob.glob(files, recursive=True)
+        for filePath in file_list:
+            subFolder = helper.emailFolder(parent_dir, filePath, True)
```
Comment (baibhavr) [inline mailbag/formats/mbox.py:41]: TypeError: emailFolder() takes 2 positional arguments but 3 were given

File: mailbag/formats/pst.py [hunk 2]
```diff
@@ -27,81 +28,131 @@ def __init__(self, target_account, args, **kwargs):
...
-                            AttachmentFiles=attachments,
-                            Error='False'
+                            AttachmentFiles=attachments
                         )
                     
                     except (Exception) as e:
                         log.error(e)
                         message = Email(
-                            Error='True'
+                            Error=['Error parsing message.']
```
Comment (baibhavr) [inline mailbag/formats/pst.py:118]: Should we also include errors from previous try exceptions into the model here?

**Review by baibhavr — CHANGES_REQUESTED** [review]

File: mailbag/formats/mbox.py (hunk 1 above)
Comment (gwiedeman) [reply mailbag/formats/mbox.py]: Fixed. This was left in from an alternate approach to `Message_Path`. Thanks for finding.

File: mailbag/formats/pst.py (hunk 2 above)
Comment (gwiedeman) [reply mailbag/formats/pst.py]: Yup, that was a bit lazy. Should be fixed now.

**Review by baibhavr — APPROVED** [review]

**Pull request merged**